spec/features/users/index_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :feature do
+  before :each do
+    @user = User.create(name: 'Ermiyas', photo: 'https://picsum.photos/200/300', bio: 'I am a software engineer',
+                        posts_counter: 0)
+    @post = Post.create(title: 'This is my first post!!', text: 'I like it!!', author_id: @user.id,
+                        comments_counter: 0, likes_counter: 0)
+  end
+
+  describe 'index page' do
+    it 'shows the username of all other users' do
+      visit users_path
+      expect(page).to have_content(@user.name)
+      expect(page).to have_selector('.user-card', count: User.count)
+    end
+
+    it 'shows the profile picture for each user.' do
+      visit users_path
+      expect(page).to have_css("img[src='#{@user.photo}']", wait: 30)
+    end
+
+    it 'shows the number of posts each user has written' do
+      visit users_path
+      expect(page).to have_text(@user.posts_counter)
```

Comment:
- [ ] **[OPTIONAL]** This test appears to be failing, I strongly recommend you fix it. This is only optional as the previous reviewer did not highlight it, but it is mandated.

  ![hfurur](https://github.com/mrprotocoll/mp_blog/assets/36542986/e8d056dc-e4a3-4cb3-a5b3-88d4a791a484)